### ISSUE TITLE
Why does loss_r_test become larger and larger when pre-training?

### ISSUE THREAD

**Issue by Kobe972** [qitianwu/IDCF#2]
I'm training ml-1m.
Part of the outputs are shown below:
```
Epoch 93 Step 76704: Train 0.3622 Reg: 0.4783
Test: 0.9648 MAE: 0.7485 RMSE: 0.9823
Val: 0.9936 MAE: 0.7561 RMSE: 0.9968
Epoch 94 Step 77520: Train 0.3618 Reg: 0.4776
Test: 0.9654 MAE: 0.7486 RMSE: 0.9826
Val: 0.9943 MAE: 0.7562 RMSE: 0.9971
Epoch 95 Step 78336: Train 0.3614 Reg: 0.4769
Test: 0.9659 MAE: 0.7484 RMSE: 0.9828
Val: 0.9948 MAE: 0.7561 RMSE: 0.9974
Epoch 96 Step 79152: Train 0.3611 Reg: 0.4763
Test: 0.9665 MAE: 0.7489 RMSE: 0.9831
Val: 0.9955 MAE: 0.7566 RMSE: 0.9978
Epoch 97 Step 79968: Train 0.3608 Reg: 0.4757
Test: 0.9673 MAE: 0.7495 RMSE: 0.9835
Val: 0.9963 MAE: 0.7572 RMSE: 0.9982
Epoch 98 Step 80784: Train 0.3605 Reg: 0.4751
Test: 0.9676 MAE: 0.7494 RMSE: 0.9837
Val: 0.9967 MAE: 0.7571 RMSE: 0.9983
Epoch 99 Step 81600: Train 0.3603 Reg: 0.4746
Test: 0.9680 MAE: 0.7494 RMSE: 0.9839
Val: 0.9970 MAE: 0.7571 RMSE: 0.9985
```
As you can see, loss_r_test grew from 0.9648 to 0.9680. In fact, it started to be abnormal after epoch 13.
```
Epoch 10 Step 8976: Train 0.7002 Reg: 0.5511
Test: 0.7185 MAE: 0.6660 RMSE: 0.8476
Val: 0.7338 MAE: 0.6722 RMSE: 0.8566
Epoch 11 Step 9792: Train 0.6809 Reg: 0.5869
Test: 0.7136 MAE: 0.6643 RMSE: 0.8447
Val: 0.7276 MAE: 0.6705 RMSE: 0.8530
Epoch 12 Step 10608: Train 0.6638 Reg: 0.6277
Test: 0.7109 MAE: 0.6614 RMSE: 0.8431
Val: 0.7244 MAE: 0.6667 RMSE: 0.8511
Epoch 13 Step 11424: Train 0.6438 Reg: 0.6702
Test: 0.7085 MAE: 0.6602 RMSE: 0.8417
Val: 0.7217 MAE: 0.6654 RMSE: 0.8495
Epoch 14 Step 12240: Train 0.6232 Reg: 0.6993
Test: 0.7114 MAE: 0.6602 RMSE: 0.8435
Val: 0.7256 MAE: 0.6661 RMSE: 0.8518
Epoch 15 Step 13056: Train 0.6061 Reg: 0.7176
Test: 0.7164 MAE: 0.6620 RMSE: 0.8464
Val: 0.7304 MAE: 0.6671 RMSE: 0.8547
```

**Comment by qitianwu** [comment]
Hi, the pretraining stage aims to obtain embeddings for key users that will used for computing inductive representations for query users in the next stage. Therefore, it is normal that the pretrained embeddings for testing are not accurate. For more details about two-stage training and the inductive CF models, please look at our paper.

**Comment by Kobe972** [comment]
Thanks for answering my question! I want to do some research based on your model and I also have the following questions:
1. There's a formula output = (ratings + x3) / 2. + user_b + item_b in your code. However, user_b for test users are not trained so it is meaningless. Could I remove user_b if I do not require more accuracy for trained users?
2. Are there any other terms that I can remove if I am willing to sacrifice some performance?

在2022-07-18 14:28:59，Qitian ***@***.***写道：

Hi, the pretraining stage aims to obtain embeddings for key users that will used for computing inductive representations for query users in the next stage. Therefore, it is normal that the pretrained embeddings for testing are not accurate. For more details about two-stage training and the inductive CF models, please look at our paper.

**Comment by qitianwu** [comment]
I would suggest doing some exploratory experiments if you are going to remove the bias term. Given the information you provide, I cannot give you some specific advices. Anyway, the attention mechanisms are actually quite time-consuming, so one potential way is to optimize its efficiency when you do not care about the performance very much
